File: frontend/src/app/models/reserve.model.ts
import { Establishment } from "./establishment.model";
import { UserModel } from "./user.model";

export interface Reserve {
    id?: number;
    idUser?: number;
    idRestaurant?: number;
    period: string;
    reserveDate: Date;
    amountOfPeople: number;
    observation?: string;
    user?: UserModel;
    restaurant?: Establishment,
    confirmed?: boolean;
}

export interface ReservePerDay {
    day: string;
    maxNumberOfPeople: number;
    reserves: Reserve[];
    currentPeople: number;
}

File: backend/src/main/java/com/temreserva/backend/temreserva_backend/web/model/response/ReserveModel.java
package com.temreserva.backend.temreserva_backend.web.model.response;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ReserveModel {
    private Long id;

    private String period;

    private LocalDateTime reserveDate;

    private Integer amountOfPeople;

    private String observation;

    private Boolean confirmed;

    private UserModel user;

    private RestaurantModel restaurant;
}
